README.md

```diff
@@ -199,16 +199,54 @@ that it uses existing drop methods as garbage collection finalisers, saving
 users the potentially error-prone task of manually writing both destructor and
 finaliser methods for GC managed objects.
 
-When a `Gc`'s underlying `GcBox`  becomes unreachable, Alloy will call its
+When a `Gc`'s underlying contents becomes unreachable, Alloy will call its
 finaliser, which means that `drop` is called on all the component types (in the
 same way that Rust automatically calls `drop` in an RAII context).
 
-#### TODO: still need to explain
+To achieve Alloy's goal of making cyclic data structures easier to write, we
+made the decision to support the finalisation of objects with cycles. This means
+that no guarantees are made about the order in which finalisers are run. In
+order for this to be sound, you must not access other `Gc` objects from inside
+the `drop` method a `Gc`. Alloy ensures that this rule is followed by checking
+for potential misuses of `drop` at compile-time in a pass known as _Finaliser
+Safety Analysis_ (FSA). Consider the following example:
+
+```rust
+struct Node {
+    next: Gc<usize>,
+}
+
+impl Drop for Node {
+    fn drop(&mut self) {
+        *self.next;
+    }
+}
+
+fn main {
+    let x = Gc::new(Node { Gc::new(123) });
+}
+```
+
+If at any point you try to create `Gc<Node>`, such as in `main` above, Alloy will throw the following
+compiler error:
+
+```
+error: `Node { next: Gc::new(123) }` cannot be safely finalized.
+  --> src/main.rs
+   |
+12 |     let x = Gc::new(Node { next: Gc::new(123) });
+   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^ has a drop method which cannot be safely finalized.
+...
+2  |         *self.next;
+   |          ---------
+   |          |
+   |          caused by the expression here in `fn drop(&mut)` because
+   |          it uses another `Gc` type.
+   |
+   = help: `Gc` finalizers are unordered, so this field may have already been dropped. It is not safe to dereferenc
```

Comment:
Unfortunate line brea

k.

Reply:
Fixed here: 29dcc937228ac01bb19a048a1b092fafd08d488a

Reply:
Not sure you've pushed that commit?

Reply:
I raced with github. Should work now